apps/chat/src/components/Common/ApplicationDialog.tsx

```diff
@@ -36,688 +33,561 @@ import Modal from '@/src/components/Common/Modal';
-  isOpen,
+const ApplicationDialogView: React.FC<Props> = ({
   onClose,
   isEdit,
-  selectedApplication,
   currentReference,
-}: Props) => {
-  const dispatch = useAppDispatch();
-  const { t } = useTranslation(Translation.PromptBar);
-  const [name, setName] = useState('');
-  const [version, setVersion] = useState('');
-  const [versionHasFocus, setVersionHasFocus] = useState(false);
-  const [featuresDataHasFocus, setFeaturesDataHasFocus] = useState(false);
-  const [description, setDescription] = useState('');
-  const [deleteLogo, setDeleteLogo] = useState(false);
-  const [localLogoFile, setLocalLogoFile] = useState<string | undefined>();
-  const files = useAppSelector(FilesSelectors.selectFiles);
-  const featuresDataInputRef = useRef<HTMLTextAreaElement>(null);
-  const [features, setFeatures] = useState('');
-  const [featuresDataError, setFeaturesDataError] = useState('');
-  const [maxAttachments, setMaxAttachments] = useState(0);
-  const [completionUrl, setCompletionUrl] = useState('');
-  const [completionUrlError, setCompletionUrlError] = useState('');
-  const [submitted, setSubmitted] = useState(false);
-  const [nameError, setNameError] = useState<string | null>(null);
-  const [versionError, setVersionError] = useState<string | null>(null);
-  const [iconError] = useState<string | null>(null);
-  const nameInputRef = useRef<HTMLInputElement>(null);
-  const descriptionInputRef = useRef<HTMLTextAreaElement>(null);
-  const [filterParams, setFilterParams] = useState<string[]>([]);
-  const [isDeleteModalOpen, setIsDeleteModalOpen] = useState(false);
-  const [isPublishing, setIsPublishing] = useState(false);
-
-  const entity = selectedApplication && {
-    name: selectedApplication.name,
-    id: ApiUtils.decodeApiUrl(
-      selectedApplication.id ||
-        (selectedApplication as unknown as { application: string }).application,
-    ),
-    folderId: getFolderIdFromEntityId(selectedApplication.name),
-  };
-
-  const resetForm = () => {
-    setName('');
-    setVersion('');
-    setDescription('');
-    setFeatures('');
-    setMaxAttachments(0);
-    setCompletionUrl('');
-    setLocalLogoFile(undefined);
-    setDeleteLogo(false);
-    setFilterParams([]);
-  };
+}) => {
+  const {
+    register,
+    handleSubmit,
+    setValue,
+    clearErrors,
+    trigger,
+    control,
+    formState: { errors, isValid },
+  } = useForm<FormData>({
+    mode: 'onChange',
+    reValidateMode: 'onChange',
+  });
 
-  const loading = useAppSelector((state) => state.application.loading);
+  const { t } = useTranslation(Translation.Chat);
 
-  const handleChangeFilterParams = useCallback((filterParams: string[]) => {
-    setFilterParams(filterParams);
-  }, []);
+  const dispatch = useAppDispatch();
 
-  const handleClose = useCallback(() => {
-    setSubmitted(false);
-    resetForm();
-    onClose(false);
-  }, [onClose]);
+  const selectedApplication = useAppSelector(
+    ApplicationSelectors.selectApplicationDetail,
+  );
```

Comment:
```suggestion
  const selectedApplication = useAppSelector(
    state => isEdit ? ApplicationSelectors.selectApplicationDetail(state) : undefined,
  );
```

Reply:
I think that it's better to don't change it here. It's doesn't really increase performance, but may reduce readability. We just select filed from redux state

Reply:
@Alexander-Kezik, ok, but in case if editing mode value in state should be `undefined`

Reply:
I think in this case it's better to pass it as a prop?